src/main/java/randoop/resource/ClassAnnotationScanner.java

```diff
@@ -0,0 +1,148 @@
+package randoop.resource;
+
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Set;
+import org.objectweb.asm.AnnotationVisitor;
+import org.objectweb.asm.ClassVisitor;
+import org.objectweb.asm.MethodVisitor;
+import org.objectweb.asm.Opcodes;
+import org.objectweb.asm.TypePath;
+import org.plumelib.reflection.Signatures;
+
+/**
+ * ClassAnnotationScanner is a ClassVisitor that collects methods that are annotated with specific
+ * annotations.
+ */
+public class ClassAnnotationScanner extends ClassVisitor {
+  /** OpCode for ASM used throughout this class. */
+  private static final int ASM_OPCODE = Opcodes.ASM7;
+
+  /** The desired annotations that serve as the criteria for methods we want to capture. */
+  private final Collection<String> desiredAnnotations;
+
+  /** The Randoop fully qualified class name of the class being visited. */
+  private String fullyQualifiedClassName = null;
+
+  /** The accumulated set of captured fully qualified method signatures. */
+  private final Set<String> matchingFullyQualifiedMethodSignatures = new HashSet<>();
+
+  /**
+   * Initializes a new AnnotationScanner to look for methods with the specified annotations.
+   *
+   * @param api API version for ASM
+   * @param desiredAnnotations annotations to look for
+   */
+  public ClassAnnotationScanner(int api, Collection<String> desiredAnnotations) {
+    super(api);
+    this.desiredAnnotations = desiredAnnotations;
+  }
+
+  /**
+   * Returns the Randoop fully qualified signature of a method. e.g. "java.util.String.hashCode()"
+   *
+   * @param method method name
+   * @param argumentSignature signature of the method's arguments, e.g.
+   *     "(char[],int,int,java.lang.String,int)"
+   * @return Randoop fully qualified signature
+   */
+  private String getFullyQualifiedMethodSignature(String method, String argumentSignature) {
+    return fullyQualifiedClassName + "." + method + argumentSignature;
+  }
+
+  /**
+   * Adds a method to the set of captured methods if any annotation on or within the method
+   * signature is in {@code desiredAnnotations}. Ignores constructors and static constructors.
+   *
+   * @param annotationName annotation name, e.g.
+   *     "org.checkerframework.checker.determinism.qual.NonDet"
+   * @param method method name
+   * @param argumentSignature signature of the method's arguments, e.g.
+   *     "(char[],int,int,java.lang.String,int)"
+   */
+  private void addMethodIfMatching(String annotationName, String method, String argumentSignature) {
+    if (method.contains("<init>") || method.contains("<clinit>")) {
+      return; // Ignore constructors and static constructors
+    }
+
+    if (desiredAnnotations.contains(
+        Signatures.fieldDescriptorToBinaryName(annotationName.trim()))) {
```

Comment:
Why is the call to `trim()` necessary?  If it's necessary, I would move it to the client.